src/common/transformations/src/transformations/convert_precision.cpp

```diff
@@ -416,7 +418,12 @@ bool fuse_type_to_matrix_nms(const std::shared_ptr<ngraph::Node>& node, ngraph::
 }
 
 bool fuse_type_to_multiclass_nms(const std::shared_ptr<ngraph::Node>& node, ngraph::element::Type to, size_t idx) {
-    auto nms = ov::as_type_ptr<opset8::MulticlassNms>(node);
+    std::shared_ptr<ov::op::util::MulticlassNmsBase> nms;
+    if (node->get_type_info() == opset8::MulticlassNms::get_type_info_static()) {
```

Comment:
`is_type<opset8::MulticlassNms>`?

Reply:
bool DiscreteTypeInfo::is_castable(const DiscreteTypeInfo& target_type) const {
    return *this == target_type || (parent && parent->is_castable(target_type));
}

Reply:
done